src/modules/roc_fec/reader.cpp

```diff
@@ -284,29 +275,25 @@ void Reader::update_source_packets_() {
             roc_panic("fec reader: unexpected non-rtp source packet");
         }
 
-        if (!packet::seqnum_lt(fec->blknum,
-                               packet::seqnum_t(cur_block_sn_ + source_block_.size()))) {
+        if (!packet::seqnum_le(fec->source_block_number, cur_sbn_)) {
```

Comment:
I think we need a separate set of helpers (blknum_diff_t, blknum_diff, blknum_le, blknum_lt) for the blknum_t. We should not rely on the fact that blknum_t and seqnum_t are the same type currently.

Reply:
Done